internal/replication/blockreplicator.go

```diff
@@ -312,6 +315,13 @@ Event_Loop:
 
 		case <-br.stopCh:
 			br.lg.Info("Stopping block replicator")
+
+			// notify the propose-loop go-routine in case it is waiting for blocks to commit
+			br.mutex.Lock()
+			br.numInFlightBlocks = 0
+			br.condTooManyInFlightBlocks.Broadcast()
```

Comment:
Should this be placed within the lock?

Reply:
`br.numInFlightBlocks = 0` must be in the lock as it accessed by a different go-routine, and `br.condTooManyInFlightBlocks.Broadcast()` can be placed either within or without (as the documentation on Signal and Broadcast explains), the result will be (almost) the same. It will wake up the waiting routine, which will try and get the lock as it exits from Wait, will block for it, and when the lock is released it will get it and discover the value change. The same can be done with a Broadcast after the Unlock(), and the waking go routine will also try to lock as it exits Wait. I prefer the former, there is no difference in safety. See:
https://go101.org/article/concurrent-synchronization-more.html